### ISSUE TITLE
Remove all uses of `fig.subplots_adjust` in example gallery

### ISSUE THREAD

**Issue by JDWarner** [scikit-image/scikit-image#1775]
See these and similar: 

- [x] ~~~http://scikit-image.org/docs/dev/auto_examples/plot_canny.html~~~ http://scikit-image.org/docs/dev/auto_examples/edges/plot_canny.html
- [x] ~~~http://scikit-image.org/docs/dev/auto_examples/plot_medial_transform.html~~~ http://scikit-image.org/docs/dev/auto_examples/edges/plot_medial_transform.html 
- [x] ~~~http://scikit-image.org/docs/dev/auto_examples/plot_watershed.html~~~ http://scikit-image.org/docs/dev/auto_examples/segmentation/plot_watershed.html

This is too verbose and complicated for examples. When necessary, minimize whitespace with `plt.tight_layout()` instead.

**Comment by stefanv** [comment]
Hrm, how about we rather set ``plt.tight_layout`` as a default plot option for the entire gallery?

**Comment by JDWarner** [comment]
:+1:

**Comment by emmanuelle** [comment]
:+1:

**Comment by emmanuelle** [comment]
@JDWarner would you prefer modifying #1806 to remove the calls to ``tight_layout`` and add ``figure.autolayout:True`` in ``doc/source/conf.py`` or would you prefer that a separate PR is made for the latter change, and rebase on it?

**Comment by JDWarner** [comment]
I can modify, but it won't be until Saturday. Feel free to take my changes and run with them if there is a release deadline.

I think I'm going to have to rebase regardless once #1810 is merged...

**Comment by stefanv** [comment]
That should be fine, thanks Josh.

**Comment by alexandrejaguar** [comment]
Hi everyone,
what is wrong on #1810? Could I help with that? If yes, how could I do?

**Comment by blink1073** [comment]
We have known problems with `pyamg` and the `matplotlib 1.5` / `PySide` combo.  See https://github.com/scikit-image/scikit-image/issues/1788 and https://github.com/scikit-image/scikit-image/issues/1802.

**Comment by emmanuelle** [comment]
On second thoughts, keeping ``plt.tight_layout()`` in the python files of the examples makes it possible to run the example file as a script and to get the same layout as in the html documentation. This feature might be nice enough to keep an additional line of code, what do you think?

**Comment by blink1073** [comment]
I agree, it should be self-contained.

**Comment by stefanv** [comment]
This is just to strip the white space around the box, isn't it? Putting
tight layout in every single example for a plot that is functionally the
same feels dirty. I already have people looking at code snippets, any then
saying "oh, no, so much code". Then I have to point out that most of it is
actually plotting related.

**Comment by blink1073** [comment]
I'd argue that matplotlib plots are hideous without `tight_layout`, and may turn off potential users who may not even know it exists.

**Comment by stefanv** [comment]
OK, how about we have a standard header for each example that does a numpy
import and setting up the plotting environment (perhaps via a style, so
that we can include much of the code in scikit-image itself)?

And why is tight layout not the matplotlib default?

**Comment by blink1073** [comment]
That seems reasonable.  I'm not sure why is not the default, but the 2.0 release is all about setting new defaults, so it might be worth raising an issue about it.

**Comment by alexandrejaguar** [comment]
Hi everyone,
some subplots requires `tight_layout` to appear free of white space. According to my tests, this happens mainly on subplots with 1 row, 3 cols. Maybe rewriting the examples would solve that.
IMHO, I believe the examples have to be simple and explained as possible. What do you think? 
Standards related to header and some variables would be also nice, I believe.

**Comment by blink1073** [comment]
It looks like there is no rc_param for tight_layout, you have to call it explicitly via `fig.tight_layout()` or `plt.tight_layout()`.  I'd like for us to use `skimage.io.imshow` in our examples, like this:

```python
import numpy as np
import matplotlib.pyplot as plt
from scipy import ndimage as ndi

from skimage import feature, io


# Generate noisy image of a square
im = np.zeros((128, 128))
im[32:-32, 32:-32] = 1

im = ndi.rotate(im, 15, mode='constant')
im = ndi.gaussian_filter(im, 4)
im += 0.2 * np.random.random(im.shape)

# Compute the Canny filter for two values of sigma
edges1 = feature.canny(im)
edges2 = feature.canny(im, sigma=3)

# display results
fig, (ax1, ax2, ax3) = plt.subplots(nrows=1, ncols=3, figsize=(8, 3), sharex=True, sharey=True)

fig.sca(ax1)
io.imshow(im)
ax1.axis('off')
ax1.set_title('noisy image', fontsize=20)

fig.sca(ax2)
io.imshow(edges1)
ax2.axis('off')
ax2.set_title('Canny filter, $\sigma=1$', fontsize=20)

fig.sca(ax3)
io.imshow(edges2)
ax3.axis('off')
ax3.set_title('Canny filter, $\sigma=3$', fontsize=20)

fig.tight_layout()

plt.show()
```

<img width="640" alt="screen shot 2015-12-12 at 12 26 49 pm" src="https://cloud.githubusercontent.com/assets/2096628/11763325/a668b312-a0cb-11e5-826b-b6c96fb35a06.png">

**Comment by JDWarner** [comment]
We could add `fig.tight_layout()` to our own `io.show()` and use that instead of plt.show()

**Comment by blink1073** [comment]
Yep, already did.

**Comment by blink1073** [comment]
Oh, I see what you mean.

**Comment by blink1073** [comment]
I had added it to `imshow`.

**Comment by blink1073** [comment]
I think that is fine, since we are not otherwise wrapping `show`.

**Comment by soupault** [comment]
So, what's the agreement here?

**Comment by tacaswell** [comment]
`tight_layout` is not the default behavior in mpl because it is not 100% reliable (there are some artists it does not know about), does not work with all ways of making axes, and requires 2 passes through the rendering to get all of the sizes right.  You can pass `tight_layout=True` as a kwarg to both `plt.figure()` and `plt.subplots()`

**Comment by jni** [comment]
@tacaswell 

> You can pass `tight_layout=True` as a kwarg to both `plt.figure()` and `plt.subplots()`

I learn something new and useful every time you drop by. =D

**Comment by soupault** [comment]
@tacaswell I always use `plt.tight_layout()` right before `figsave()` or `plt.show()` in order to get fancy spacing. Although I don't really utilize any extraordinary capabilities of `matplotlib` (`twinx`, `twiny`, long ticklabels), that is for the reasons you described above.
As far as we use only basic plotting routines in `skimage` gallery, this change should be sufficient.

**Comment by soupault** [comment]
All originally requested cases are fixed.
Few more are:

- [ ] `./examples/xx_applications/plot_coins_segmentation.py` (in two places), http://scikit-image.org/docs/dev/auto_examples/applications/plot_coins_segmentation.html
- [ ] `./source/plots/hough_tf.py` <- this to be removed?
- [ ] `./tools/plot_pr.py`
